sound/soc/sof/ipc4-topology.c

```diff
@@ -1032,6 +1077,52 @@ static int sof_ipc4_update_hw_params(struct snd_sof_dev *sdev, struct snd_pcm_hw
 	return 0;
 }
 
+static void
+sof_ipc4_update_cpc_from_tplg(struct snd_sof_dev *sdev,
+			      struct snd_sof_widget *swidget,
+			      struct sof_ipc4_module_params *module_params,
+			      struct sof_ipc4_base_module_cfg *basecfg,
+			      bool exact_match)
+{
+	struct sof_ipc4_tplg_mod_cfg *mod_cfgs = module_params->mod_cfgs;
+	u32 cpc_pick = 0;
+	u32 max_cpc = 0;
+	int i;
+
+	/*
+	 * No mod_cfg present for the module or the CPC is already set, no need
+	 * to proceed further
+	 */
+	if (!mod_cfgs || basecfg->cpc)
+		return;
+
+	for (i = 0; i < module_params->num_mod_cfgs; i++) {
+		if (basecfg->obs == mod_cfgs[i].obs &&
+		    basecfg->ibs == mod_cfgs[i].ibs &&
+		    cpc_pick < mod_cfgs[i].cpc)
+			cpc_pick = mod_cfgs[i].cpc;
```

Comment:
I've seen that code before and it's still wrong - and it's not clear why we copy/pasted. can a helper be used?

Reply:
We have different struct for manifest_mod_cfg and for the tplg_mod_cfg.
The tplg version only contains ibs/obs/cpc, while the manifest have more fields (is_bytes/cps/ibs/obs/module_flags/cpc/obls)

Reply:
I still thin the code is OK.